From 3f14df87160831daaa187abe27bc9440bc4b0bb0 Mon Sep 17 00:00:00 2001
From: Jon Anderson <jon.anderson@joyent.com>
Date: Fri, 8 Mar 2019 12:22:17 -0500
Subject: [PATCH] MANTA-4150 Muppet test case writeHaproxyConfig fails on
 SmartOS Reviewed by: Kelly McLaughlin <kelly.mclaughlin@joyent.com> Approved
 by: Kelly McLaughlin <kelly.mclaughlin@joyent.com>

---
 package.json        |  5 +++--
 test/config.test.js | 30 +++++++++++++++++++-----------
 2 files changed, 22 insertions(+), 13 deletions(-)

diff --git a/package.json b/package.json
index a9b24e3..c54fb9f 100644
--- a/package.json
+++ b/package.json
@@ -10,8 +10,8 @@
         "bunyan": "2.0.2",
         "dashdash": "1.14.1",
         "forkexec": "1.1.0",
+        "haproxy-stat": "0.1.0",
         "jsprim": "1.4.0",
-        "haproxy-stat" : "0.1.0",
         "node-uuid": "1.4.1",
         "once": "1.4.0",
         "vasync": "2.2.0",
@@ -19,7 +19,8 @@
         "zkstream": "0.11.3"
     },
     "devDependencies": {
-        "nodeunit": "0.11.2"
+        "nodeunit": "0.11.2",
+        "diff": "4.0.1"
     },
     "scripts": {
         "start": "node ./muppet.js"
diff --git a/test/config.test.js b/test/config.test.js
index 0f0aba7..6d50618 100644
--- a/test/config.test.js
+++ b/test/config.test.js
@@ -7,13 +7,12 @@
 /*
  * Copyright (c) 2017, Joyent, Inc.
  */
-
 var vasync = require('vasync');
-var execFile = require('child_process').execFile;
 var helper = require('./helper.js');
 var lbm = require('../lib/lb_manager.js');
 var path = require('path');
 var fs = require('fs');
+var jsdiff = require('diff');
 
 ///--- Globals
 var test = helper.test;
@@ -110,15 +109,24 @@ test('test writeHaproxyConfig', function (t) {
     };
     lbm.writeHaproxyConfig(opts, function (err, data) {
         t.equal(null, err);
-        // compare output file with reference
-        execFile('diff', ['-wq', '-I', 'log-send-hostname',
-                          updConfig_out, updConfig_out_chk],
-            function (error, _stdout, _stderr) {
-                t.equal(null, error);
-                // cleanup test file
-                fs.unlinkSync(updConfig_out);
-                t.done();
-            });
+        var test_txt = fs.readFileSync(updConfig_out, 'utf8');
+        var check_txt = fs.readFileSync(updConfig_out_chk, 'utf8');
+
+        var diff = jsdiff.diffTrimmedLines(test_txt, check_txt);
+
+        diff.forEach(function (part) {
+            if (part.added) {
+                if (! part.value.includes('log-send-hostname')) {
+                    t.equal(null, part.value);
+                }
+            } else if (part.removed) {
+                if (! part.value.includes('log-send-hostname')) {
+                    t.equal(null, part.value);
+                }
+            }
+        });
+        fs.unlinkSync(updConfig_out);
+        t.done();
     });
 });
 
-- 
2.21.0

